spec/lib/auth0/api/v2/resource_servers_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+describe Auth0::Api::V2::ResourceServers do
+  before :all do
+    dummy_instance = DummyClass.new
+    dummy_instance.extend(Auth0::Api::V2::ResourceServers)
+    dummy_instance.extend(Auth0::Mixins::Initializer)
+    @instance = dummy_instance
+  end
+
+  context '.resource_server' do
+    it { expect(@instance).to respond_to(:resource_server) }
+    it 'is expected to call get /api/v2/resource-servers/test' do
+      expect(@instance).to receive(:get).with('/api/v2/resource-servers/test')
+      expect { @instance.resource_server('test') }.not_to raise_error
+    end
+    it 'expect to raise an error when calling with empty resource server id' do
+      expect { @instance.resource_server(nil) }.to raise_error 'Must supply a valid resource server id'
+    end
+  end
+
+  context '.create_resource_server' do
+    it { expect(@instance).to respond_to(:create_resource_server) }
+    it 'is expected to call post /api/v2/resource-servers' do
+      expect(@instance).to receive(:post).with(
+        '/api/v2/resource-servers',
+        identifier: 'test',
+        name: 'name',
+        signing_alg: 'signing_alg',
+        signing_secret: 'signing_secret',
+        token_lifetime: 'token_lifetime',
+        scopes: 'scopes')
+
+      @instance.create_resource_server('test', name: 'name',
+                                               signing_alg: 'signing_alg',
+                                               signing_secret: 'signing_secret',
+                                               token_lifetime: 'token_lifetime',
+                                               scopes: 'scopes')
+    end
+    it 'expect to raise an error when calling with empty identifier' do
+      expect { @instance.create_resource_server(nil) }.to raise_error 'Must supply a valid resource server id'
+    end
```

Comment:
should we test creating a resource server with name containing "<". It should return error as the character is not supported